utils/httpUtils.ts

```diff
@@ -72,3 +71,114 @@ export const notFound = (message: string) => {
     message: message,
   });
 };
+
+export const getFeatureFlagsFromRequest = context => {
+  return context.request.headers.get("x-should-read-from-nextgen");
+};
+
+export const shouldReadFromNextGen = async context => {
+  let shouldReadFromNextGen = getFeatureFlagsFromRequest(context);
+  if (shouldReadFromNextGen === true || shouldReadFromNextGen === "true" || shouldReadFromNextGen === "True") {
+    // if the header is set, return the value
+    return true;
+  }
+  return false;
+};
+
+export const formatFedQueryForNextGen = (query: string) => {
+  let cleanQuery = query;
+  // remove fed Prefix
+  const firstCurlyBracket = query.indexOf("{");
+  const secondCurlyBracket = query.indexOf("{", firstCurlyBracket + 1);
+  const fedIndex = query.indexOf("fed");
+  if (fedIndex !== -1 && fedIndex < secondCurlyBracket && fedIndex > firstCurlyBracket) {
+    const splitOnFirstFed = [query.substring(0, fedIndex + 4), query.substring(fedIndex + 4)];
+    const splitQueryOnFed = splitOnFirstFed[0].split("fed");
+    const firstLetterLowerCase = splitQueryOnFed[1][0].toLowerCase();
+    splitQueryOnFed[1] = splitQueryOnFed[1].slice(1);
+    splitQueryOnFed.splice(1, 0, firstLetterLowerCase);
+    cleanQuery = splitQueryOnFed.join("").concat(splitOnFirstFed[1]);
+  }
+
+  // remove input object from variables
+  const finishedQuery = cleanQuery
+    .replace("input: {", "")
+    .replace("}}", "}")
+    // apply any specific type switches that need to be made - these can be passed in from the resolver
+    .replace("$workflowRunId: String", "$workflowRunId: UUID") // lets make an actual UUID
+    .replace(/query_fedConsensusGenomes_items/g, "ConsensusGenome");
+
+  return finishedQuery;
+};
+
+export const fetchFromNextGen = async ({
+  args,
+  context,
+  serviceType,
+  fullResponse,
+  customQuery,
+}: {
+  args;
+  context;
+  serviceType: "workflows" | "entities";
+  fullResponse?: boolean;
+  customQuery?: string;
+}) => {
+  const enrichedToken = await getEnrichedToken(context);
+  const baseUrl = serviceType === "workflows" ? process.env.NEXTGEN_WORKFLOWS_URL : process.env.NEXTGEN_ENTITIES_URL;
+  const formattedQuery = customQuery ? customQuery : formatFedQueryForNextGen(context.params.query);
+  const response = await fetch(`${baseUrl}/graphql`, {
+    method: "POST",
+    headers: {
+      Cookie: context.request.headers.get("cookie"),
+      "Content-Type": "application/json",
+      "X-CSRF-Token": args?.input?.authenticityToken,
+      Authorization: `Bearer ${enrichedToken}`,
+    },
+    body: JSON.stringify({
+      query: formattedQuery,
+      variables: context.params.variables,
+    }),
+  });
+  if (fullResponse === true) {
+    console.log("full response", response);
```

Comment:
these console logs might be redundant with some of the other ones?

Reply:
All these console logs we mostly for debugging. Would it be best to remove them now?

Reply:
maybe best to delete? I don't care that much though, if you don't feel like it just leave it :P